File: Projects/Clock/Clock/Counter.cs
// Student Name: Nguyen Xuan Nang Mai. Student ID: 105549980.
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Clock
{
    public class Counter
    {
        private int _count;
        private string _name;

        public Counter(string name)
        {
            _name = name;
            _count = 0;
        }

        public void Increment()
        {
            _count++;
        }

        public void Reset()
        {
            _count = 0;
        }

        public string Name
        {
            get
            {
                return _name;
            }
            set
            {
                _name = value;
            }

        }

        public int Ticks
        {
            get
            {
                return _count;
            }
        }

        public void ResetByDefault()
        {
            try
            {
                checked
                {
                    _count = 0;
                }
            }
            catch (OverflowException ex)
            {
                Console.WriteLine($"Overflow error: {ex.Message}");
            }
        }
    }
}

File: Projects/ClockPython/Counter.py
class Counter:
    def __init__(self, name):
        self._name = name
        self._count = 0

    def Increment(self):
        self._count += 1

    def Reset(self):
        self._count = 0

    def ResetByDefault(self):
        try:
            self._count = 0
        except OverflowError as ex:
            print(f"Overflow error: {ex}")

    @property
    def Name(self):
        return self._name

    @Name.setter
    def Name(self, value):
        self._name = value

    @property
    def Ticks(self):
        return self._count
    